src/S3/MalformedResponseParser.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Api\Parser\AbstractParser;
+use Aws\CommandInterface;
+use Aws\Exception\AwsException;
+use Exception;
+use Psr\Http\Message\ResponseInterface;
+
+class MalformedResponseParser extends AbstractParser
+{
+    /** @var callable */
+    private $parser;
+    /** @var string */
+    private $exceptionClass;
+
+    public function __construct(
+        callable $parser,
+        $exceptionClass = AwsException::class
+    ) {
+        $this->parser = $parser;
+        $this->exceptionClass = $exceptionClass;
+    }
+
+    public function __invoke(
+        CommandInterface $command,
+        ResponseInterface $response
+    ) {
+        $fn = $this->parser;
+
+        try {
+            return $fn($command, $response);
+        } catch (Exception $e) {
```

Comment:
Unfortunately, SimpleXMLElement will only throw one class of exception: `Exception`. It seemed convenient, as a generic `MalformedResponseParser` could be used with any service, not just S3, but for that to be true I think I would need to define an `Aws\ParserException` and translate JSON and SimpleXML errors into that.